### ISSUE TITLE
Question Abount Backends

### ISSUE THREAD

**Issue by Mrs-Feathers** [alan-turing-institute/apricot#9]
Hello! I came across this software and its exactly what i need to get linux LDAP login working as all the rest of my network runs off of OIDC and i wanted to extend that to linux, but PAM and SSSD only support LDAP not OIDC.. so this project is perfect for that. however, i'm not using a microsoft server.. and i was wondering what other backends you support. i think its just a default OpenID Connect server.. in specific the software i'm using is Authentik (and yes it does say it supports LDAP but it's too buggy and doesn't support full LDAP... but the OIDC is absolutely amazing out of the box.)

so, basically, is there more documentation about which backends you support?

**Comment by jemrobinson** [comment]
@Mrs-Feathers : Currently this is still in an alpha state. I've only written a Microsoft Entra backend because that's what I'm using. Are you interested in working on adding an Authentik backend? This should be relatively simple to do, but I don't have time for it at present.

The project is open for PRs if you do have some time for this.

**Comment by tctlrd** [comment]
Will this work with Zitadel as OIDC backend?

**Comment by jemrobinson** [comment]
@tctlrd : there isn't an existing Zitadel backend plugin, but you're welcome to contribute one.

**Comment by ChristianJacobsen** [comment]
@tctlrd @jemrobinson, I'm working on a Zitadel backend now. Hopefully I have something to review in a few days. Didn't look like a whole lot of work, but I'm unfamiliar with both LDAP and Zitadel, so this is "best effort" :)

**Comment by ChristianJacobsen** [comment]
I've tried to get a hold of the Zitadel devs in their Discord to answer some questions about "hard-to-query resources", but they haven't gotten back to me as of writing this message.

If I've understood Zitadel correctly, the only place you can create custom "groups" (which I think is "roles" in Zitadel speak) is through their "projects". For a given project you can create custom roles which you can assign (authorize) users to.

They (Zitadel) are in the middle of an API upgrade, and it seems the new API will allow someone to query users authorized to a certain role, which seems to be what we need for the `groups()` query in the `OAuthClient`. I haven't been able to find an API resource in the current V1 API that yields this information.

We could of course make this organization-scoped instead and then just use the fixed roles assigned to users when they are added to the organization, but that leaves to way to create custom roles that map to group labels expected by other applications that require LDAP to function.

IMHO it's better to wait with the new backend until the V2 API has been completed.

**Comment by tfmm** [comment]
Zitadel's v2 API is now GA as of this morning - https://zitadel.com/docs/apis/v2 - Unfortunately I do not have the knowledge to contribute updates to the code here.

**Comment by ChristianJacobsen** [comment]
> Zitadel's v2 API is now GA as of this morning - https://zitadel.com/docs/apis/v2 - Unfortunately I do not have the knowledge to contribute updates to the code here. 

Then I know what I'm doing tomorrow morning 😊 Thanks for letting me know!
